core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -628,12 +565,12 @@ public void onConnectionClosedByServer(Status status) {
 
                 @Override
                 public void onInUse(TransportSet ts) {
-                  inUseStateAggregator.updateObjectInUse(ts, true);
+                  isIdle = false;
                 }
 
                 @Override
                 public void onNotInUse(TransportSet ts) {
-                  inUseStateAggregator.updateObjectInUse(ts, false);
+                  isIdle = true;
```

Comment:
I only messed with this since it didn't compile after removing `inUseStateAggregator`.

I don't know the requirements of what has to happen exactly, (or how a transport exits idle mode).  

@zhangkun83 thoughts?